tasks/code_sync_status.rb

```diff
@@ -0,0 +1,95 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+# Puppet Task Name: code_sync_status
+require 'net/https'
+require 'uri'
+require 'json'
+require 'puppet'
+
+# CodeSyncStatus task class
+class CodeSyncStatus
+  def initialize(params)
+    @params = params
+  end
+
+  def execute!
+    puts sync_status.to_json
+  end
+
+  private
+
+  def https_client
+    client = Net::HTTP.new('localhost', '8140')
+    client.use_ssl = true
+    client.cert = @cert ||= OpenSSL::X509::Certificate.new(File.read(Puppet.settings[:hostcert]))
+    client.key = @key ||= OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
+    client.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    client
+  end
+
+  def api_status
+    status = https_client
+    # Only debug level includes code sync details
+    status_request = Net::HTTP::Get.new('/status/v1/services?level=debug')
+    JSON.parse(status.request(status_request).body)
+  end
+
+  def check_environment_list(environments, request_environments)
+    environmentstocheck = []
+    # If all was passed as an argument we check all visible environments
+    if request_environments.any? { |s| s.casecmp('all') == 0 }
+      environmentstocheck = environments
+    # Else check each requested environment to confirm its a visible environment
+    else
+      request_environments.each do |environment|
+        environments.any? { |s| s.casecmp(environment.to_s) == 0 } || raise("Environment #{environment} is not visible and will not be checked")
+        environmentstocheck << environment
+      end
+    end
+    environmentstocheck
+  end
+
+  def check_environment_code(environment, servers, status_call)
+    # Find the commit ID of the environment according to the file sync service
+    primarycommit = status_call['file-sync-storage-service']['status']['repos']['puppet-code']['submodules'][environment.to_s]['latest_commit']['message'][32..71]
```

Comment:
```suggestion
    primarycommit = status_call.dig('file-sync-storage-service',
                                    'status',
                                    'repos',
                                    'puppet-code',
                                    'submodules',
                                    environment.to_s,
                                    'latest_commit',
                                    'message').split("'").last
```

Reply:
I don't know if this actually works, but it's an example of another approach for getting the value you want out of the data structure, and then munging the piece of the string you want out of it.

Reply:
This works perfectly great suggestion @reidmv